### PR TITLE
Allow eggs to be mixed together just as we would with alcohol.

### PR DESCRIPTION
**Pull request by kwl01skz**
#### Summary
None

#### Purpose of change

It's hard to collect enough identical eggs to meet the requirements of the recipe.
I'm trying to handle this in a similar way to "mixed alcohol", just using JSON.  It makes eggs easier to utilize overall.

#### Describe the solution

I've added a recipe that allows players to mix different types of bird eggs or reptilian eggs together.

#### Describe alternatives you've considered

We could add insect eggs to the recipe.
But that doesn't seem smart. 1 insect egg = 12 bird eggs. I can't just modify the JSON to make things easy.
So I hesitate about this.
I think we can simplify this in C++ code in the future.

```

  {
    "type": "recipe",
    "activity_level": "NO_EXERCISE",
    "result": "rehydrated_eggs",
    "id_suffix": "from_egg_3x",
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_OTHER",
    "time": "30 s",
    "charges": 3,
    "batch_time_factors": [ 80, 2 ],
    "//": "mixed eggs like mixed alcohol.",
    "autolearn": true,
    "flags": [ "BLIND_EASY" ],
    "components": [
      [
        [ "egg_roach", 1 ],
        [ "egg_roach_plague", 1 ],
        [ "egg_locust", 1 ],
        [ "egg_dragonfly", 1 ],
        [ "egg_firefly", 1 ],
        [ "egg_wasp", 1 ],
        [ "egg_strider", 1 ],
        [ "egg_grasshopper", 1 ],
        [ "egg_lady_bug", 1 ],
        [ "egg_mole_cricket", 1 ],
        [ "egg_water_beetle", 1 ],
        [ "egg_water_bug", 1 ],
        [ "egg_butterfly", 1 ],
        [ "egg_moth", 1 ],
        [ "egg_cicada", 1 ],
        [ "egg_fly", 1 ],
        [ "egg_mosquito", 1 ],
        [ "eggs_spider", 1, "LIST" ], 
        [ "egg_snail", 1 ], 
        [ "egg_slug", 1 ]
      ]
    ]
  },
  {
    "type": "recipe",
    "activity_level": "NO_EXERCISE",
    "result": "rehydrated_eggs",
    "id_suffix": "from_egg_6x",
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_OTHER",
    "time": "60 s",
    "charges": 6,
    "batch_time_factors": [ 80, 2 ],
    "//": "mixed eggs like mixed alcohol.",
    "autolearn": true,
    "flags": [ "BLIND_EASY" ],
    "components": [
      [
        [ "egg_roach", 2 ],
        [ "egg_roach_plague", 2 ],
        [ "egg_locust", 2 ],
        [ "egg_dragonfly", 2 ],
        [ "egg_firefly", 2 ],
        [ "egg_wasp", 2 ],
        [ "egg_strider", 2 ],
        [ "egg_grasshopper", 2 ],
        [ "egg_lady_bug", 2 ],
        [ "egg_mole_cricket", 2 ],
        [ "egg_water_beetle", 2 ],
        [ "egg_water_bug", 2 ],
        [ "egg_butterfly", 2 ],
        [ "egg_moth", 2 ],
        [ "egg_cicada", 2 ],
        [ "egg_fly", 2 ],
        [ "egg_mosquito", 2 ],
        [ "eggs_spider", 2, "LIST" ], 
        [ "egg_snail", 2 ], 
        [ "egg_slug", 2 ],
        [ "egg_antlion", 1 ],
        [ "egg_centipede", 1 ]
      ]
    ]
  },
  {
    "type": "recipe",
    "activity_level": "NO_EXERCISE",
    "result": "rehydrated_eggs",
    "id_suffix": "from_egg_12x",
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_OTHER",
    "time": "120 s",
    "charges": 12,
    "batch_time_factors": [ 80, 2 ],
    "//": "mixed eggs like mixed alcohol.",
    "autolearn": true,
    "flags": [ "BLIND_EASY" ],
    "components": [ [ [ "eggs_invert", 1, "LIST" ] ] ]
  },

```

#### Additional context

#45157 
#64546

### CODE REVIEW

File: data/json/items/comestibles/egg.json [hunk 1]
```diff
@@ -742,21 +742,22 @@
   {
     "type": "COMESTIBLE",
     "id": "rehydrated_eggs",
-    "copy-from": "egg_bird",
-    "name": { "str": "rehydrated egg" },
+    "name": { "str_sp": "liquid eggs" },
```
Comment (Fris0uman) [inline data/json/items/comestibles/egg.json:745]: why overriding rehydrated eggs here? Just make a new item, it's weird that mixing eggs makes rehydrated eggs when they should be made from egg powder

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -4932,24 +4933,16 @@
   },
   {
     "type": "recipe",
-    "activity_level": "LIGHT_EXERCISE",
+    "activity_level": "NO_EXERCISE",
     "result": "fried_egg_sandwich",
```
Comment (Fris0uman) [inline data/json/recipes/recipe_food.json:4937]: this could probably do with using the suffix infrastructure to have one recipe where you cook the egg and make everything from scratch and one where you use already fried eggs. Forcing players to do the fried egg recipe first and then the sadnwich recipe adds a tiny bit of annoyance that we could do without

File: data/json/recipes/recipe_food.json [hunk 3]
```diff
@@ -4967,24 +4960,16 @@
   },
   {
     "type": "recipe",
-    "activity_level": "LIGHT_EXERCISE",
+    "activity_level": "NO_EXERCISE",
```
Comment (Fris0uman) [inline data/json/recipes/recipe_food.json:4963]: same thing, this could be 2 recipes

File: data/json/recipes/recipe_food.json [hunk 4]
```diff
@@ -4892,13 +4892,14 @@
...
-    "time": "4 m",
+    "time": "2 m 30 s",
     "autolearn": true,
     "batch_time_factors": [ 80, 2 ],
+    "charges": 1,
     "qualities": [ { "id": "COOK", "level": 2 } ],
-    "tools": [ [ [ "surface_heat", 7, "LIST" ] ] ],
+    "tools": [ [ [ "surface_heat", 4, "LIST" ] ] ],
     "//": "6u for the egg, 1u for the oil",
-    "components": [ [ [ "eggs_bird", 2, "LIST" ], [ "egg_reptile", 2 ] ], [ [ "any_butter_or_oil", 2, "LIST" ] ] ]
```
Comment (Fris0uman) [inline data/json/recipes/recipe_food.json:4901]: this seems completly out of scope for this PR, if you want to change requirement for a recipe do it in a different PR with the accompanying justifications

**Review by Fris0uman — COMMENTED** [review]
Pulling eggs in a generic egg requirement is probably a good idea, I don't get the whole rehydrated egg renamed into liguid egg part though

File: data/json/items/comestibles/egg.json (hunk 1 above)
Comment (kwl01skz) [reply data/json/items/comestibles/egg.json:745]: I once heard from food factories that rehydrated eggs contain all the nutrients of eggs except the shell.  Rehydrated eggs are (egg powder + water) = all the nutrients of the egg liquid, which is the same as the nutrition obtained by directly crushing an egg.  Some time ago, I added the item "rehydrated eggs" to allow the protagonist to mix different kinds of eggs.  But I found that this didn't quite serve the purpose – because making whole egg flour itself requires the same kind of eggs, so this change was made.

File: data/json/recipes/recipe_food.json (hunk 4 above)
Comment (kwl01skz) [reply data/json/recipes/recipe_food.json:4901]: The goal of this PR is to give us the freedom to use any type of egg.
But the old recipe called for two identical eggs, so I changed it.

File: data/json/recipes/recipe_food.json (hunk 2 above)
Comment (kwl01skz) [reply data/json/recipes/recipe_food.json:4937]: Adding many of the same recipes is something I'd like to avoid.
Just like the code I didn't add above.

Now, recipes of the same product (different suffixes) are piled up, and I can't simplify them, and I can't fold them.
I hope that in the future, players will be allowed more freedom to use alternative ingredients, rather than adding massive products to the same recipe. (like PR), which requires C++ code to implement.

Comment (kwl01skz) [reply data/json/recipes/recipe_food.json:4937]: Many sandwich recipes, both in the game and in real life, actually go through a two-step process: take something that can't be eaten raw and make it edible, then put it into slices of bread. Like "meat sandwiches".

So, I'm not actually subverting player behavior. Just made the fried egg sandwiches more like meat sandwiches.

Comment (kwl01skz) [reply data/json/recipes/recipe_food.json:4937]: Many sandwich recipes, both in the game and in real life, actually go through a two-step process: take something that can't be eaten raw and make it edible, then put it into slices of bread. Like "meat sandwiches".

So, I'm not actually subverting player behavior. Just made the fried egg sandwiches more like meat sandwiches.

File: data/json/recipes/recipe_food.json (hunk 3 above)
Comment (kwl01skz) [reply data/json/recipes/recipe_food.json:4963]: I look forward to the day in the future when we can remove all wheat-free recipes.
I think we can do this by simply adding a flag like 'CANNIBALISM'.

Maybe it's up to me to try to deal with that future - in a few months when I have nothing else to do.

**Review by LyleSY — APPROVED** [review]
What's here makes sense. There's going to be way more work in DinoMod to properly take advantage of that, but I'm happy to manage that bit.

File: data/json/items/comestibles/egg.json [hunk 5]
```diff
@@ -742,21 +742,22 @@
...
-    "price_postapoc": 25,
+    "description": "Liquid eggs that contains almost all the nutrients of eggs.  If you accidentally drop the eggshell fragments into it, it may be able to add a little extra calcium.",
     "material": [ "egg" ],
     "volume": "50 ml",
-    "flags": [  ],
-    "fun": -2,
+    "stack_size": 1,
+    "fun": -8,
+    "flags": [ "FREEZERBURN", "RAW" ],
+    "vitamins": [ [ "calcium", 3 ], [ "iron", 4 ] ],
```
Comment (Fris0uman) [inline data/json/items/comestibles/egg.json:759]: Are we really considering this thing to contain eggshells?

File: data/json/recipes/recipe_food.json (hunk 2 above)
Comment (Fris0uman) [reply data/json/recipes/recipe_food.json:4937]: Adding a recipe is really not much of a problem, they could even fit in a recipe category so that they can be collapsed in the menu

File: data/json/recipes/recipe_food.json [hunk 6]
```diff
@@ -8855,7 +8855,7 @@
     "tools": [ [ [ "surface_heat", 6, "LIST" ] ] ],
     "//": "3u for egg, 2u for milk, 1u for alcohol/sugar",
     "components": [
-      [ [ "eggs_bird", 1, "LIST" ], [ "egg_reptile", 1 ], [ "powder_eggs", 1 ] ],
```
Comment (Fris0uman) [inline data/json/recipes/recipe_food.json:8858]: Same question as above

**Review by Fris0uman — COMMENTED** [review]
A few more tweaks

File: data/json/items/comestibles/egg.json (hunk 1 above)
Comment (Maleclypse) [inline data/json/items/comestibles/egg.json:745]:
```suggestion
    "name": { "str_sp": "rehydrated eggs" },
```
In America liquid eggs mean the below product which has removed the yolks.  https://www.target.com/p/cage-free-liquid-egg-substitute-32oz-good-38-gather-8482/-/A-79519411?ref=tgt_adv_xsp&AFID=google&fndsrc=tgtao&DFA=71700000108264730&CPNG=PLA_Dairy%2BFrozen_Local%7CDairy%2BFrozen_Ecomm_Food_Bev&adgroup=SC_Dairy%2BFrozen&LID=700000001170770pgs&LNM=PRODUCT_GROUP&network=g&device=c&location=1025952&targetid=pla-511168984938&gclid=CjwKCAjw9pGjBhB-EiwAa5jl3BrIMTPjR99SwyNro12Bwy94BM96s6FtMRhv1zyCqjiEd_y8Wu6zWRoCO2MQAvD_BwE&gclsrc=aw.ds

File: data/json/items/comestibles/egg.json [hunk 7]
```diff
@@ -742,21 +742,22 @@
...
-    "symbol": "%",
-    "looks_like": "powder_egg",
-    "quench": 2,
+    "symbol": "~",
+    "quench": 4,
     "calories": 80,
-    "description": "Rehydrated eggs.  You can use them instead of eggs when cooking, provided you don't care that they have lost their complete shape.",
-    "price": 22,
-    "price_postapoc": 25,
+    "description": "Liquid eggs that contains almost all the nutrients of eggs.  If you accidentally drop the eggshell fragments into it, it may be able to add a little extra calcium.",
```
Comment (Maleclypse) [inline data/json/items/comestibles/egg.json:753]:
```suggestion
    "description": "Rehydrated eggs.  You can use them instead of eggs when cooking, provided you don't care that they have lost their complete shape.",
```

File: data/json/items/comestibles/egg.json (hunk 5 above)
Comment (Maleclypse) [reply data/json/items/comestibles/egg.json:759]: Eggs contain some calcium without eggshells.

**Review by Maleclypse — CHANGES_REQUESTED** [review]
The only changes I think make sense in this PR is putting  eggs_any_shape   into lists that previously had bird and reptile eggs in them and don't require the egg to be in a specific shape, such as fried or sunny side up.   Rehydrated eggs include the yolk liquid eggs don't.  Fried eggs sandwich I noticed both you and the heating audit missed that the bread is also fried in the oil for a second.
